src/meson.build

```diff
@@ -417,7 +425,7 @@ defs_list += [harfbuzz_subset_def]
 
 libharfbuzz_subset = library('harfbuzz-subset', hb_subset_sources,
   include_directories: incconfig,
-  dependencies: deps,
+  dependencies: [],
```

Comment:
Looks very good, thank you very much!

I had to put this to get `libm` from deps, https://github.com/harfbuzz/harfbuzz/commit/bb095e1bdd0f7f1de88651d6f7223bc04bf70d70 in C linker which needs [meson master](https://github.com/mesonbuild/meson/commit/1bfeaadd6d1dc87532a1c0ad4be615eb34044495#diff-69c59a112064153b548fa7e9d432af44) (not released yet 0.55) to test, is `dependencies: [],` makes it somehow work? Thanks

Reply:
> I had to put this to get `libm` from deps, [bb095e1](https://github.com/harfbuzz/harfbuzz/commit/bb095e1bdd0f7f1de88651d6f7223bc04bf70d70) as it isn't done automatically in C linker which needs [meson master](https://github.com/mesonbuild/meson/commit/1bfeaadd6d1dc87532a1c0ad4be615eb34044495#diff-69c59a112064153b548fa7e9d432af44) (not released yet 0.55) to test, is `dependencies: [],` makes it somehow work?

I didn't find any libm usage there, but I'll have another look.

> And can we drop empty list of `dependencies: [],` in other places?

Sure, I just left it as "documentation" in case it is needed in the future.

Reply:
Steps to reproduce:

```
# install meson master
pip install git+https://github.com/mesonbuild/meson
# the revision before that change
git checkout dca8ba6b
# build using meson
rm -rf build && meson build && ninja -Cbuild test
```

```
FAILED: src/libharfbuzz-subset.so.0.20605.0 
clang  -o src/libharfbuzz-subset.so.0.20605.0 'src/25a6634@@harfbuzz-subset@sha/hb-number.cc.o' 'src/25a6634@@harfbuzz-subset@sha/hb-ot-cff1-table.cc.o' 'src/25a6634@@harfbuzz-subset@sha/hb-ot-cff2-table.cc.o' 'src/25a6634@@harfbuzz-subset@sha/hb-static.cc.o' 'src/25a6634@@harfbuzz-subset@sha/hb-subset-cff-common.cc.o' 'src/25a6634@@harfbuzz-subset@sha/hb-subset-cff1.cc.o' 'src/25a6634@@harfbuzz-subset@sha/hb-subset-cff2.cc.o' 'src/25a6634@@harfbuzz-subset@sha/hb-subset-input.cc.o' 'src/25a6634@@harfbuzz-subset@sha/hb-subset-plan.cc.o' 'src/25a6634@@harfbuzz-subset@sha/hb-subset.cc.o' -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,--start-group -Wl,-soname,libharfbuzz-subset.so.0 src/libharfbuzz.so.0.20605.0 -Wl,--end-group '-Wl,-rpath,$ORIGIN/' -Wl,-rpath-link,~/harfbuzz/build/src
/usr/bin/ld: src/25a6634@@harfbuzz-subset@sha/hb-ot-cff1-table.cc.o: in function `hb_font_t::em_scalef(float, int)':
~/harfbuzz/build/../src/hb-font.hh:625: undefined reference to `roundf'
/usr/bin/ld: src/25a6634@@harfbuzz-subset@sha/hb-subset.cc.o: in function `_plan_estimate_subset_table_size(hb_subset_plan_t*, unsigned int)':
~/harfbuzz/build/../src/hb-subset.cc:64: undefined reference to `sqrt'
clang-10: error: linker command failed with exit code 1 (use -v to see invocation)
```

But am not able to reproduce using current state of the build system :)

Reply:
> But am not able to reproduce using current state of the build system for some reason :)

Okay, now I see, my fault :/ https://github.com/harfbuzz/harfbuzz/commit/d67553d12fc72c4a89fcb75400829b62d53ca064 please rebase to this, thanks

Reply:
I've added `m_dep` back now and remove one remaining empty dependency argument.